### PR TITLE
add USE_USERNAME config

### PR DESCRIPTION
**Pull request by samkuehn**
I have an exiting user database that stores username as just username (not email address).  This added a  USE_USERNAME option to the config to allow for this.

### CODE REVIEW

**Comment by AzMoo** [comment]
Hey thanks so much for your contribution, but I can't accept a change like this without tests or docs.

**Comment by samkuehn** [comment]
Sorry for not adding tests or docs.  I had not used poetry before and was being lazy.  Test and docs have been added

**Review by AzMoo — CHANGES_REQUESTED** [review]
Please don't commit a `poetry.lock`, we don't need it for a library.

File: okta_oauth2/tokens.py [hunk 1]
```diff
@@ -92,12 +92,14 @@ def handle_token_result(self, token_result):
         if claims:
             tokens["id_token"] = token_result["id_token"]
             tokens["claims"] = claims
-
+            username = claims["email"]
+            if self.config.use_username:
+                username = claims["email"].split("@")[0]
```
[the remark this answers is not in the document]
Comment (samkuehn) [reply okta_oauth2/tokens.py:97]: Interesting.  Never seen an email with two @'s in it.  Do you thing we need to account for this?

**Review by AzMoo — APPROVED** [review]

**Comment by samkuehn** [comment]
Any chance we could get this change in released to pypi?

**Comment by AzMoo** [comment]
Yep, sorry for the delay. I didn't realise the CI had a problem with the code style.

**Pull request merged**